### ISSUE TITLE
Add "Julia" to modular interfaces

### ISSUE THREAD

**Issue by karlnapf** [shogun-toolbox/shogun#1612]
http://julialang.org/

Julia is a pretty hip young open-source language for computing.
It is as easy as Matlab and almost as fast as C (at least much faster than python/R/octave). It has a growing community and I know quite a few people that are already using it, even though it is very young and not yet complete.

If we offered Machine Learning to this young community, this would be very cool since nothing similar yet exists so more reasons to use Shogun. Julia can also interface with the ipython notebooks - so people could write Julia notebooks.

The community is very active, maybe we even could get some help.

What do you think?

**Comment by ChrisRackauckas** [comment]
Since this was discussed, Clang.jl and Cxx.jl have improved immensely and make this form of semi-auto wrapping much easier. Now that this is part of NumFocus, I think this should be reconsidered!

**Comment by karlnapf** [comment]
Absolutely, but what we need here is the ability of SWIG to link to Julia. This should not be too hard to be done by someone knowing SWIG and julia a bit .... @ChrisRackauckas you know someone?
